static/js/unread.js

```diff
@@ -264,9 +264,55 @@ exports.num_unread_for_person = function (user_ids_string) {
     return unread_privates.get(user_ids_string).num_items();
 };
 
+exports.current_stream_topic = function () {
+    var selected_stream;
+    var selected_topic;
+    if (narrow.filter() === undefined || narrow.filter()._operators["1"] === undefined) {
+         var message_header = $(".selected_message").closest(".recipient_row").find(".message_header");
+         selected_stream = ($(message_header).find(".stream_label").html()).replace(/ |\n/g,'');
+         selected_topic = ($(message_header).find(".stream_topic a").text()).replace(/ |\n/g,'');
+    } else {
+         selected_stream = narrow.filter()._operators["0"].operand;
+         selected_topic = narrow.filter()._operators["1"].operand;
+    }
+
+    return { stream : selected_stream, topic : selected_topic };
+};
+
+// Returns the next unread topic,
+// To select the selected topic pass select_this_topic as true
+exports.next_topic = function (selected_stream, selected_topic) {
+    // Get ordered stream list`
+    stream_list.build_stream_list();
+    var streams = stream_list.stream;
+
+    var topic_res = exports.unread_topic_counter.get_counts().subject_count;
+
+    // the topic list already contains only those which are unread
+    function has_unread_messages() {
+        return true;
+    }
+
+    function get_topics(stream) {
+        var topic_list = [];
+        if (topic_res._items[':' + stream] === undefined) {
```

Comment:
It looks like you are peeking into the implementation of the Dict class!  I am guessing you figured out this code by poking around with the console, which is a good technique to some degree, but you should read `dict.js`! :)